website/server/controllers/top-level/payments/iap.js

```diff
@@ -85,6 +89,123 @@ api.iapAndroidVerify = {
   },
 };
 
+/**
+ * @apiIgnore Payments are considered part of the private API
+ * @api {post} /iap/android/subscription Android Subscribe
+ * @apiName IapAndroidSubscribe
+ * @apiGroup Payments
+ **/
+api.iapsubscriptionAndroid = {
+  method: 'POST',
+  url: '/iap/android/subscribe',
+  middlewares: [authWithUrl],
+  async handler (req, res) {
+    let subCode;
+    switch (req.body.sku) {
+      case 'com.habitrpg.android.habitica.subscription.1month':
+        subCode = 'basic_earned';
+        break;
+      case 'com.habitrpg.android.habitica.subscription.3month':
+        subCode = 'basic_3mo';
+        break;
+      case 'com.habitrpg.android.habitica.subscription.6month':
+        subCode = 'basic_6mo';
+        break;
+      case 'com.habitrpg.android.habitica.subscription.12month':
+        subCode = 'basic_12mo';
+        break;
+    }
+    let sub = subCode ? shared.content.subscriptionBlocks[subCode] : false;
+
+    if (!req.body.sku) throw new BadRequest(res.t('missingSubscriptionCode'));
+
+    let user = res.locals.user;
+    let iapBody = req.body;
+
+
+    await iap.setup();
+
+    let testObj = {
+      data: iapBody.transaction.receipt,
+      signature: iapBody.transaction.signature,
+    };
+
+    let receiptObj = JSON.parse(testObj.data); // passed as a string
+    let token = receiptObj.token || receiptObj.purchaseToken;
+
+    let existingUser = await User.findOne({
+      'payments.plan.customerId': token,
+    }).exec();
+    if (existingUser) throw new NotAuthorized('RECEIPT_ALREADY_USED');
```

Comment:
@TheHollidayInn this is fine because @vIiRuS is using error.message as a way to signal an error code to the app and not a readable message to the users, so there's no need to translate it